src/config/SearchContext.tsx

```diff
@@ -0,0 +1,157 @@
+import { createContext, useCallback, useEffect, useState } from 'react';
+import { BackHandler } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+
+import ApiConstants from '@network/apiConstants';
+import { _get } from '@network/instanceMethods';
+import { debounce } from '@utility/helpers';
+
+const defaultValue: SearchContextValues = {
+  data: [],
+  searchText: '',
+  onChangeText: () => {},
+  onSubmit: () => {},
+  suggestions: [],
+  showSuggestions: false,
+  cuisines: [],
+  diets: [],
+  ingredients: [],
+  handleCuisine: () => {},
+  handleDiet: () => {},
+  handleIngredient: () => {},
+  onBackPress: () => true,
+};
+
+const SearchContext = createContext<SearchContextValues>(defaultValue);
+
+export const SearchContextProvider = ({ children }: ContextProviderProps) => {
+  const {
+    data: defaultData,
+    searchText: defaultSearch,
+    suggestions: defaultAutoCompletedQueries,
+    showSuggestions: defaultShowAutoComplete,
+    cuisines: defaultCuisines,
+    diets: defaultDiets,
+    ingredients: defaultIngredients,
+  } = defaultValue;
+
+  const { goBack } = useNavigation<StackNavigation>();
+
+  const [data, setData] = useState<SearchRecipes>(defaultData);
+  const [searchText, setSearchText] = useState(defaultSearch);
+  const [searchedOn, setSearchedOn] = useState('');
+  const [suggestions, setSuggestions] = useState<AutoCompleteSearches>(defaultAutoCompletedQueries);
+  const [showSuggestions, setShowSuggestions] = useState(defaultShowAutoComplete);
+
+  const [cuisines, setCuisines] = useState<Cuisines>(defaultCuisines);
+  const [ingredients, setIngredients] = useState<Ingredients>(defaultIngredients);
+  const [diets, setDiets] = useState<Diets>(defaultDiets);
+
+  const [loading, setLoading] = useState(false);
+
+  const { autoCompleteSearch: autoCompleteEndpoint, complexSearch: complexSearchEndpoint } =
+    ApiConstants.endpoints;
+
+  const fetchAutoComplete = async (query: string) => {
+    const autoCompleteResponse = await _get<AutoCompleteSearches>(autoCompleteEndpoint, {
+      params: { query, number: 1 },
+    });
+
+    if (autoCompleteResponse.success) {
+      const { responseData } = autoCompleteResponse;
+      setSuggestions(responseData);
+      setShowSuggestions(true);
+    }
+  };
+
+  const debouncedSearch = useCallback(debounce(fetchAutoComplete, 600), []);
+
+  const onBackRequest = () => {
+    if (data.length && showSuggestions) setShowSuggestions(false);
```

Comment:
use like this 
    if (data.length && showSuggestions) 
    {
setShowSuggestions(false)
}

for better understanding